doc/examples/uuid.rst

```diff
@@ -0,0 +1,472 @@
+     - :class:`uuid.UUID` in PyMongo<4; :class:`~bson.binary.Binary` subtype 4 in PyMongo>=4
+     - :class:`uuid.UUID`
+
+   * - :ref:`standard-representation-details`
+     - No
+     - :class:`~bson.binary.Binary` subtype 4
+     - :class:`uuid.UUID`
+     - :class:`uuid.UUID` in PyMongo<4; :class:`~bson.binary.Binary` subtype 3 in PyMongo>=4
+
+   * - :ref:`unspecified-representation-details`
+     - Yes, in PyMongo>=4
+     - Raise :exc:`ValueError`
+     - :class:`~bson.binary.Binary` subtype 4
+     - :class:`uuid.UUID` in PyMongo<4; :class:`~bson.binary.Binary` subtype 3 in PyMongo>=4
+
+We now detail the behavior and use-case for each supported UUID
+representation.
+
+.. _python-legacy-representation-details:
+
+``PYTHON_LEGACY``
+^^^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used when reading UUIDs
+   generated by existing applications that use the Python driver
+   but **don't** explicitly set a UUID representation.
+
+.. attention:: :data:`~bson.binary.UuidRepresentation.PYTHON_LEGACY`
+   has been the default uuid representation since PyMongo 2.9.
+
+The :data:`~bson.binary.UuidRepresentation.PYTHON_LEGACY` representation
+corresponds to the legacy representation of UUIDs used by PyMongo. This
+representation conforms with
+`RFC 4122 Section 4.1.2 <https://tools.ietf.org/html/rfc4122#section-4.1.2>`_.
+
+The following example illustrates the use of this representation::
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import UuidRepresentation
+
+  # No configured UUID representation
+  collection = client.python_legacy.get_collection('test', codec_options=DEFAULT_CODEC_OPTIONS)
+
+  # Using UuidRepresentation.PYTHON_LEGACY
+  pylegacy_opts = CodecOptions(uuid_representation=UuidRepresentation.PYTHON_LEGACY)
+  pylegacy_collection = client.python_legacy.get_collection('test', codec_options=pylegacy_opts)
+
+  # UUIDs written by PyMongo with no UuidRepresentation configured can be queried using PYTHON_LEGACY
+  uuid_1 = uuid4()
+  collection.insert_one({'uuid': uuid_1})
+  document = pylegacy_collection.find_one({'uuid': uuid_1})
+
+  # UUIDs written using PYTHON_LEGACY can be read by PyMongo with no UuidRepresentation configured
+  uuid_2 = uuid4()
+  pylegacy_collection.insert_one({'uuid': uuid_2})
+  document = collection.find_one({'uuid': uuid_2})
+
+``PYTHON_LEGACY`` encodes native :class:`uuid.UUID` objects to
+:class:`~bson.binary.Binary` subtype 3 objects, preserving the same
+byte-order as :attr:`~uuid.UUID.bytes`::
+
+  from bson.binary import Binary
+
+  document = collection.find_one({'uuid': Binary(uuid_2.bytes, subtype=3)})
+  assert document['uuid'] == uuid_2
+
+.. _java-legacy-representation-details:
+
+``JAVA_LEGACY``
+^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used when reading UUIDs
+   written to MongoDB by the Java driver without an explicitly configured UUID
```

Comment:
This wording makes it seem like all Java apps have this problem. Should we clarify that it's only legacy Java apps (which don't use STANDARD)?

Reply:
Done.